eland/dataframe.py

```diff
@@ -844,32 +848,31 @@ def info(
 
         lines = [str(type(self)), self._index_summary()]
 
-        if len(self.columns) == 0:
+        cols: pd.Index = self.columns
+        col_count: int = len(cols)
+
+        if col_count == 0:
             lines.append(f"Empty {type(self).__name__}")
             fmt.buffer_put_lines(buf, lines)
             return
 
-        cols = self.columns
-        col_count = len(self.columns)
-
         # hack
         if max_cols is None:
-            max_cols = pd.get_option("display.max_info_columns", len(self.columns) + 1)
+            max_cols = pd.get_option("display.max_info_columns", col_count + 1)
 
         max_rows = pd.get_option("display.max_info_rows", len(self) + 1)
 
-        if null_counts is None:
-            show_counts = (len(self.columns) <= max_cols) and (len(self) < max_rows)
-        else:
-            show_counts = null_counts
-        exceeds_info_cols = len(self.columns) > max_cols
+        if show_counts is None:
+            show_counts = (col_count <= max_cols) and (len(self) < max_rows)
+
+        exceeds_info_cols = col_count > max_cols
 
         # From pandas.DataFrame
         def _put_str(s, space) -> str:
             return f"{s}"[:space].ljust(space)
 
         def _verbose_repr() -> None:
-            lines.append(f"Data columns (total {len(self.columns)} columns):")
+            lines.append(f"Data columns (total {len(cols)} columns):")
```

Comment:
Can we use `number_of_columns` here? Also add `nonlocal number_of_columns` at the top of the function.